src/ec/packages/ec-component-radio/radio.story.js

```diff
@@ -8,46 +9,140 @@ import dataBinary from './demo/data--binary';
 import radioGroup from './ecl-radio-group.html.twig';
 import notes from './README.md';
 
+// Labels for the groups.
+const requiredGroupId = 'Mandatory elements';
+const optionalGroupId = 'Optional elements';
+const statesGroupId = 'States';
+
+const prepareRadio = data => {
+  data.invalid = boolean('invalid', false, statesGroupId);
+  data.required = boolean('required', data.required, statesGroupId);
+  data.label = text('label', data.label, requiredGroupId);
+  if (data.invalid) {
+    data.invalid_text = text(
+      'invalid_text',
+      data.invalid_text,
+      requiredGroupId
+    );
+  } else {
+    data.invalid_text = text(
+      'invalid_text',
+      data.invalid_text,
+      optionalGroupId
+    );
+  }
+  if (data.required) {
+    data.required_text = text(
+      'required_text',
+      data.required_text,
+      requiredGroupId
+    );
+    data.optional_text = text(
+      'optional text',
+      data.optional_text,
+      optionalGroupId
+    );
+  } else {
+    data.required_text = text(
+      'required_text',
+      data.required_text,
+      optionalGroupId
+    );
+    data.optional_text = text(
+      'optional_text',
+      data.optional_text,
+      requiredGroupId
+    );
+  }
+  data.helper_text = text('helper_text', data.helper_text, optionalGroupId);
+
```

Comment:
The only thing about this component, is when you pass in a special character and you find, the unicode in the html. for all the rest the component  looks perfect

Reply:
you're right, but this looks like a more general issue we have when we pass the json to the notes tab, it's not only about this knob. Trying to find a solution but no luck yet, think we can live with this for now if we don't find a straightforward solution, though.

Reply:
I've found actually something that works, but as i mentioned this is not only about this single knob, so it's something we have to fix in the notes plugin.